django/db/migrations/executor.py

```diff
@@ -120,9 +129,14 @@ def _migrate_all_forwards(self, plan, full_plan, fake, fake_initial):
                         self.progress_callback("render_success")
                 state = self.apply_migration(state, migration, fake=fake, fake_initial=fake_initial)
                 migrations_to_run.remove(migration)
-            else:
+            elif migration in applied_migrations:
```

Comment:
The `full_plan` can include migrations that are not applied or planned to be if the `targets` != `self.loader.graph.leaf_nodes()`.

For example, if we call `migrate contenttypes` without any migrations applied the full plan will include migrations excluded by the target specifier (`contenttypes`). In this case we simply want to ignore these as they shouldn't be present in the resulting state that is meant to be dispatched to `post_migrate` receivers.